consolidate_task.py

```diff
@@ -92,81 +90,63 @@ def consolidate(self):
 
         # process layers
         layers = QgsProject.instance().mapLayers()
-        self.rangeChanged.emit(len(layers))
+        self.progressRange = len(layers)
+        self.setProgress(1.0)
 
         # keep full paths of exported layer files (used to zip files)
         outFiles = [self.projectFile]
+        if self.isCanceled():
+            raise Exception('Consolidation canceled')
 
-        for layer in layers.values():
+        for i, layer in enumerate(layers.values()):
             if not layer.isValid():
-                self.processError.emit(
-                    "Layer %s is invalid" % layer.name())
-                return
+                raise Exception("Layer %s is invalid" % layer.name())
+            lType = layer.type()
+            lProviderType = layer.providerType()
+            lName = layer.name()
+            lUri = layer.dataProvider().dataSourceUri()
+            if lType == QgsMapLayer.VectorLayer:
+                # Always convert to GeoPackage
+                outFile = self.convertGenericVectorLayer(
+                    e, layer, lName)
+                outFiles.append(outFile)
+            elif lType == QgsMapLayer.RasterLayer:
```

Comment:
why don't we put this also in the geopackage?

Reply:
I've added a FIXME about the need to check this

Reply:
be careful that here we are not managing any raster layer, but only the ones defined as GDAL virtual layers to expose TMS as rasters in the map, using an external XML. This part isn't needed anymore and we should instead extend it to any raster